packages/lib/shared/pages/PoolCreationPage.tsx

```diff
@@ -4,21 +4,29 @@ import { PoolCreationForm } from '@repo/lib/modules/pool/actions/create/PoolCrea
 import { DefaultPageContainer } from '@repo/lib/shared/components/containers/DefaultPageContainer'
 import { TransactionStateProvider } from '@repo/lib/modules/transactions/transaction-steps/TransactionStateProvider'
 import { Permit2SignatureProvider } from '@repo/lib/modules/tokens/approvals/permit2/Permit2SignatureProvider'
-import { PoolCreationFormProvider } from '@repo/lib/modules/pool/actions/create/PoolCreationFormProvider'
+import {
+  PoolCreationFormProvider,
+  usePoolCreationForm,
+} from '@repo/lib/modules/pool/actions/create/PoolCreationFormProvider'
 import { TokenBalancesProvider } from '@repo/lib/modules/tokens/TokenBalancesProvider'
 import { useTokens } from '@repo/lib/modules/tokens/TokensProvider'
-import { PROJECT_CONFIG } from '@repo/lib/config/getProjectConfig'
 import { GqlChain } from '@repo/lib/shared/services/api/generated/graphql'
-import { useParams } from 'next/navigation'
 import { TokenInputsValidationProvider } from '@repo/lib/modules/tokens/TokenInputsValidationProvider'
 import { PriceImpactProvider } from '@repo/lib/modules/price-impact/PriceImpactProvider'
 
 export default function PoolCreationPage() {
+  return (
+    <PoolCreationFormProvider>
+      <PoolCreationPageContent />
+    </PoolCreationFormProvider>
+  )
+}
+
+function PoolCreationPageContent() {
   const { getTokensByChain, isLoadingTokens } = useTokens()
-  const params = useParams()
+  const { network } = usePoolCreationForm()
 
-  const selectedNetwork = (params.network as string) || PROJECT_CONFIG.defaultNetwork
-  const initTokens = getTokensByChain(selectedNetwork.toUpperCase() as GqlChain)
+  const initTokens = getTokensByChain(network.toUpperCase() as GqlChain)
```

Comment:
With this solution, I am having to refresh the page before my user token balances will show up

To test use a wallet with assets on multiple networks then
1. choose first network 
2. select token and see if it shows tokens in wallet
3. go back and choose different network
4. select token and see if it shows tokens in wallet for the newly selected network

Reply:
added a refetch of the token balances on network change